java/code/src/com/redhat/rhn/domain/action/ActionFactory.java

```diff
@@ -906,6 +914,27 @@ public static void updateServerActions(Action actionIn, List<Long> serverIds, Ac
         serverIds.forEach(sid -> SystemManager.updateSystemOverview(sid));
     }
 
+    /**
+     * Mark queue server actions as failed because the execution has been rejected
+     * @param actionsId list of ids of the action to reject
+     * @param rejectionReason the reason why the scheduled action was not picked up
+     */
+    public static void rejectScheduledActions(List<Long> actionsId, String rejectionReason) {
+        Query<Long> query = getSession().createNamedQuery("Action.rejectAction", Long.class)
+                                            .setParameter("rejection_reason", rejectionReason)
+                                            .setParameter("completion_time", new Date());
+
+        List<Long> updatedServerIds =
+            splitAndExecuteQuery(actionsId, "action_ids", query, query::list, new ArrayList<>(), (a, b) -> {
+                List<Long> result = new ArrayList<>(a);
```

Comment:
this cant be replaced with`ListUtils::union` which does the same.

Reply:
Yes, indeed I fixed it. I did not use it in the beginning because it led to an unchecked assignment, as `ListUtils::union` does not use generics. I just found out that if I declare explicitly the generic type the unchecked warning goes away :man_shrugging: